packages/compiler/src/core/name-resolver.ts

```diff
@@ -175,6 +179,11 @@ export function createResolver(program: Program): NameResolver {
   mutate(globalNamespaceNode).symbol = globalNamespaceSym;
   mutate(globalNamespaceSym.exports).set(globalNamespaceNode.id.sv, globalNamespaceSym);
 
+  /**
+   * Tracking the symbols that are used through using.
+   */
+  const usedUsingSym = new Map<TypeSpecScriptNode, Set<Sym>>();
```

Comment:
can this not just directly track the using instead of having an extra looping

Reply:
Extra work would be needed in the resolver if we want to calculate the using directly there. But because the unused-using diagnostic is optional and won't be enabled by default in cli, so I think it's better to save the work until we have to do the calculation in the linter. thx.